lib/Sema/CSDiagnostics.cpp

```diff
@@ -2963,8 +2965,11 @@ ContextualFailure::getDiagnosticFor(ContextualTypePurpose context,
 bool TupleContextualFailure::diagnoseAsError() {
   auto diagnostic = isNumElementsMismatch()
                         ? diag::tuple_types_not_convertible_nelts
-                        : diag::tuple_types_not_convertible;
-  emitDiagnostic(getAnchor()->getLoc(), diagnostic, getFromType(), getToType());
+                        : getDiagnosticFor(getContextualTypePurpose(), false);
```

Comment:
```suggestion
                        : getDiagnosticFor(getContextualTypePurpose(),
                                           /*forProtocol=*/ false);
```
Sorry for random comment in your PR, but maybe this nit make sense :))

Reply:
Thanks! I'll add this in